Client/game_sa/gamesa_init.h

```diff
@@ -104,3 +104,42 @@ void MemOrFast(U ptr, const T value)
 
 bool GetDebugIdEnabled(uint uiDebugId);
 void LogEvent(uint uiDebugId, const char* szType, const char* szContext, const char* szBody, uint uiAddReportLogId = 0);
+
+namespace CallingConventions
+{
+    struct thiscall{};
+    struct ___cdecl{};
+    struct stdcall{};
+    struct fastcall{};
+    struct vectorcall{};
+} // namespace CallingConventions
+
+template <typename ReturnType, typename CallingConvention, typename Func, typename... Args>
+typename std::enable_if<std::is_same<CallingConvention, CallingConventions::thiscall>::value, ReturnType>::type CallGTAFunction(Func function, Args... params)
+{
+    return reinterpret_cast<ReturnType(__thiscall*)(Args...)>(function)(params...);
+}
+
+template <typename ReturnType, typename CallingConvention, typename Func, typename... Args>
+typename std::enable_if<std::is_same<CallingConvention, CallingConventions::___cdecl>::value, ReturnType>::type CallGTAFunction(Func function, Args... params)
```

Comment:
There are nice suffixes _t and _v:
`typename std::enable_if_t<std::is_same_v<CallingConvention, CallingConventions::thiscall>, ReturnType> CallGTAFunction(Func function, Args&&... params)`

Reply:
Is there any difference between a function without a prefix like std::is_same and with a prefix like std::is_same_v?

Reply:
Not, just a syntactic sugar

Reply:
Instead of std::is_same<a,b>::value you can write std::is_same_v<a,b>.

Reply:
> Is there any difference between a function without a suffix like std::is_same and with a suffix like std::is_same_v?

`std::is_same_v` is just an alias to `std::is_same::value`:
```cpp
template <typename T, typename Y>
constexpr auto is_same_v = is_same<T, Y>::value;
```